src/main/java/tatracker/ui/MainWindow.java

```diff
@@ -234,15 +258,43 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleHelp();
             }
 
+            if (commandResult.isSwitchTab()) {
+                handleGoto(parseTab(commandText));
+            }
+
             if (commandResult.isExit()) {
                 handleExit();
             }
```

Comment:
So these can be moved to your parseTab method. Then the method can be renamed doNextAction(commandResult.getNextAction())

Where getNextAction() returns an enum.